.clang-tidy

```diff
@@ -1,10 +1,11 @@
 ---
 # TODO: enable more checks, run 'clang-tidy -list-checks -checks="*"' to see more
-Checks:          '-*,google-global-names-in-headers,modernize-loop-convert,readability-simplify-boolean-expr,readability-inconsistent-declaration-parameter-name'
+Checks:          '-*,google-global-names-in-headers,modernize-loop-convert,readability-simplify-boolean-expr,readability-inconsistent-declaration-parameter-name,readability-identifier-naming,cppcoreguidelines-pro-type-member-init,bugprone-*, -bugprone-macro-parentheses, -bugprone-lambda-function-name, performance-*'
```

Comment:
For our reference, can you add in the description which check does what?
For example, `readability-identifier-naming` will check for CamelCase.

Also, just confirming if the hyphen at the start of `-bugprone-macro-parentheses` and `-bugprone-lambda-function-name` is actually needed.  Couldn't find examples that include the hyphen.

Reply:
There are many checks, it would be hard to document otherwise, https://releases.llvm.org/7.0.0/tools/clang/tools/extra/docs/clang-tidy/checks/list.html

Kindly go through this list.

The hyphen in beginning is to disable those checks